src/integrationTest/java/com/mongodb/hibernate/IdentifierIntegrationTests.java

```diff
@@ -0,0 +1,250 @@
+    void endingWithDoubleQuote() {
+        var item = new EndingWithDoubleQuote();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionEndingWithDoubleQuote.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(EndingWithDoubleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(EndingWithDoubleQuote.class, item.id));
+    }
+
+    @Override
+    public void injectSessionFactoryScope(SessionFactoryScope sessionFactoryScope) {
+        this.sessionFactoryScope = sessionFactoryScope;
+    }
+
+    @Entity
+    @Table(name = WithSpaceAndDotMixedCase.COLLECTION_NAME)
+    static class WithSpaceAndDotMixedCase {
+        static final String COLLECTION_NAME = "collection with space and .dot Mixed Case";
+        static final String FIELD_NAME = "field with space Mixed Case";
+
+        @Id
+        int id;
+
+        @Column(name = WithSpaceAndDotMixedCase.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = InSingleQuotesMixedCase.COLLECTION_NAME)
+    static class InSingleQuotesMixedCase {
+        static final String COLLECTION_NAME = "`collection in single quotes Mixed Case`";
+        static final String FIELD_NAME = "`field in single quotes Mixed Case`";
+        static final String ACTUAL_COLLECTION_NAME = "collection in single quotes Mixed Case";
+        static final String ACTUAL_FIELD_NAME = "field in single quotes Mixed Case";
+
+        @Id
+        int id;
+
+        @Column(name = InSingleQuotesMixedCase.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = StartingWithSingleQuote.COLLECTION_NAME)
+    static class StartingWithSingleQuote {
+        static final String COLLECTION_NAME = "`collection starting with single quote";
+        static final String FIELD_NAME = "`field starting with single quote";
+
+        @Id
+        int id;
+
+        @Column(name = StartingWithSingleQuote.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = EndingWithSingleQuote.COLLECTION_NAME)
+    static class EndingWithSingleQuote {
+        static final String COLLECTION_NAME = "collection ending with single quote`";
+        static final String FIELD_NAME = "field ending with single quote`";
+
+        @Id
+        int id;
+
+        @Column(name = EndingWithSingleQuote.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = InDoubleQuotesMixedCase.COLLECTION_NAME)
+    static class InDoubleQuotesMixedCase {
+        static final String COLLECTION_NAME = "\"collection in double quotes Mixed Case\"";
+        static final String FIELD_NAME = "\"field in double quotes Mixed Case\"";
+        static final String ACTUAL_COLLECTION_NAME = "collection in double quotes Mixed Case";
+        static final String ACTUAL_FIELD_NAME = "field in double quotes Mixed Case";
```

Comment:
Unfortunately, it seems impossible to make identifiers like `\"collection in double quotes Mixed Case\"` to not be stripped from quotes by Hibernate ORM. Apparently, it has logic somewhere that ignores any configuration/overrides related to quoting, and strips double and single quotes anyway.

Reply:
That place seems to be here https://github.com/hibernate/hibernate-orm/blob/0a4d551b32801ae15e99d2f8ac0750303ecda2b6/hibernate-core/src/main/java/org/hibernate/boot/model/naming/Identifier.java#L159-L173.

Reply:
yeah, it seems so. If the Hibernate quoting at the start and end is hard-coded or not configurable, I think we could remove those `MixedCases` testing cases for we don't need to test Hibernate if its behaviour is working as expected.